src/modules/background/background.ts

```diff
@@ -1,33 +1,75 @@
 import { TraktAuth } from '../../api/TraktAuth';
+import { TraktScrobble } from '../../api/TraktScrobble';
+import { TraktItem } from '../../models/TraktItem';
 import { BrowserStorage, StorageValuesOptions } from '../../services/BrowserStorage';
 import { Errors } from '../../services/Errors';
 import { RequestDetails, Requests } from '../../services/Requests';
 import { Shared } from '../../services/Shared';
-import { Tabs } from '../../services/Tabs';
-import { streamingServices } from '../../streaming-services';
+import { streamingServices } from '../../streaming-services/streaming-services';
 
-interface MessageRequest {
-	action: 'check-login' | 'finish-login' | 'login' | 'logout' | 'send-request';
-	url: string;
+export type MessageRequest =
+	| {
+			action:
```

Comment:
Probably why a lot of people don't like TypeScript, but I prefer defining every single one of these as Interfaces. It will be a bit more overhead, but I think think it looks cleaner.

Reply:
I agree. It also makes it easier in the future if we need to add a property to an action.